test/lifecycle_test.dart

```diff
@@ -876,24 +872,6 @@ void sharedLifecycleTests<T extends react.Component>({
       });
     }
 
-    if (isComponent2) {
-      test('throws when state is initialized in both init and getInitialState', () {
```

Comment:
Don't we need a test asserting that calling `getInitialState` throws in `Component2`?

Reply:
@greglittlefield-wf  is going to add the throws tests for `getInitialState` and `getDefaultProps`